brightcove/brightcove-popup/src/page/Popup.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useEffect } from 'react';
+
+import { ConfigObj } from '../model/config.model';
+import { VideoList } from '../model/videoList.model';
+import { PopupObject } from '../model/receiverObj.model';
+import Modal from '../components/modal';
+
+const Popup: React.FC = () => {
+  const [message, setMessage] = useState('');
+  const [config, setConfig] = useState<ConfigObj>();
+  const [selectedVideosList, setSelectedVideosList] = useState<VideoList[]>();
+
+  useEffect(() => {
+    if (window.opener) {
+      window.opener.postMessage(
+        {
+          message: 'get Brightcove config and selected videos',
+          getConfig: true,
+        },
+        '*'
+      );
+      console.info('message :' + message);
+    } else {
+      window.close();
+    }
+
+    const receiveMessage = (event: PopupObject) => {
+      const {
+        data: { message, config, selectedVideos },
+      } = event;
+      if (config) {
+        setMessage(message);
+        setConfig(config);
+      }
+      if (selectedVideos) {
+        setMessage(message);
+        setSelectedVideosList(selectedVideos);
+      }
+    };
+    window.addEventListener('message', receiveMessage, false);
+  }, []);
+
+  const onCloseWindow = (selectedVideos: VideoList[]) => {
+    if (selectedVideos.length > 0 && window.opener) {
+      window.opener.postMessage(
+        {
+          message: 'sending selected videos',
+          selectedVideosList: selectedVideos,
+        },
+        '*'
+      );
+    } else if (window.opener) {
+      window.opener.postMessage(
+        {
+          message: 'Window closed sending selected videos',
+        },
+        '*'
+      );
+    }
+    window.close();
+  };
+
+  return (
+    <div>
+      {config && selectedVideosList ? (
```

Comment:
Are we checking `selectedVideosList` or `selectedVideosList.length` here ??
Because if we checking `selectedVideosList` then it always returns `true`
and if we checking `selectedVideosList.length` then it returns `true` when `selectedVideosList` has length greater than zero.

Reply:
here selectedVideosList may contain undefined value during initial rendering as we are setting value after initial rendering in useEffect. So to prevent undefined values to pass we need to have conditional check over here. We can't use .length as undefine.length will break the app